### PR TITLE
refactor: blackbox testing

### PR DESCRIPTION
**Pull request by friedrichwilken**
**Description**
Because of the changes in test data the validation test does no longer work. This PR will changed this.

Changes proposed in this pull request:

- Handle a error where an expectation could not be assigned and would just be dropped.
- Data for evaluation and validation into one common directory per scenario; 
     - validation data exists into `./data/namespace-scoped/<scenario name>/validation.yml`
     - evaluation data exists into `./data/namespace-scoped/<scenario name>/scenario.yml`
- Validation data is aggregated by evaluation scenario
- Added a unit test that ensures data integrity of validation test data.
- Added retry logic for validation test for situations where we run into `rate limit` issues.
**Related issue(s)**

### CODE REVIEW

**Comment by friedrichwilken** [comment]
And it works:
image

File: tests/blackbox/src/evaluation/scenario/scenario.py [hunk 1]
```diff
@@ -187,7 +187,7 @@ def load_all_namespace_scope_scenarios(self, path: str, logger: Logger) -> None:
 
         # loop over all the directory names
         for dir_name in directories:
-            scenario_file = path + "/" + dir_name + "/scenario.yml"
+            scenario_file = path + "/" + dir_name + "/evaluation.yml"
```
Comment (mfaizanse) [inline tests/blackbox/src/evaluation/scenario/scenario.py:190]: (nit): I would prefer to keep this file name as `scenario.yml`, as it more resembles as a test case.

File: tests/blackbox/src/test_validation_data.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+from validation.scenario_mock_responses import ValidationScenario
+from validation.utils.data_loader import load_data
+from validation.utils.utils import get_expectation
+
+# Unit tests run from the rootdir of the repository.
+DEFAULT_DATA_DIR: str = "tests/blackbox/data/namespace-scoped"
+
+
+def test_data_integrity():
```
Comment (mfaizanse) [inline tests/blackbox/src/test_validation_data.py:9]: Can you add comment to explain the purpose of this test?

Comment (friedrichwilken) [reply tests/blackbox/src/test_validation_data.py:9]: will do

File: tests/blackbox/src/evaluation/scenario/scenario.py (hunk 1 above)
Comment (friedrichwilken) [reply tests/blackbox/src/evaluation/scenario/scenario.py:190]: That works for me as well. Will undo the changes.

**Comment by friedrichwilken** [comment]
The validation test now also handles rate limits with retries
```
2024-10-08 18:46:37 - INFO - validation.validator : Validating expectation: complete_yaml
2024-10-08 18:46:38 - WARNING - validation.validator : Rate limit error: Error code: 429 - {'error': 'TooManyRequest', 'message': 'Your request has been rate limited by AI Core. Please try again later.'}. Retrying in 2 minutes...
2024-10-08 18:48:39 - INFO - validation.validator : Validating expectation: deployable_yaml
```

**Review by mfaizanse — DISMISSED** [review]

**Review by mfaizanse — DISMISSED** [review]

File: tests/blackbox/README.md [hunk 3]
```diff
@@ -2,9 +2,17 @@
 
 The black box testing framework provides evaluation tests for Kyma Companion and validation of large language models (LLMs) as assessments for tests. The evaluation tests measure the improvement of Kyma Companion, while validation finds the best model to verify the test scenarios.
 
+## Testing Data
+
+The data we use for both `Validation` and `Evaluation` is stored in `kyma-companion/tests/blackbox/data/namespace-scoped`. Located in that directery are a number of subdirectories named after the testing scenario they represent. In each of these you will find:
```
Comment (mmitoraj) [inline tests/blackbox/README.md:7]:
```suggestion
The data we use for both `Validation` and `Evaluation` is stored in `kyma-companion/tests/blackbox/data/namespace-scoped`. Located in that directory are several subdirectories named after the testing scenario they represent. In each of these you will find:
```

File: tests/blackbox/README.md [hunk 4]
```diff
@@ -2,9 +2,17 @@
...
+- `deploy.sh`, a shell script that would help to deploy the scenario on a Kubernetes cluster.
```
Comment (mmitoraj) [inline tests/blackbox/README.md:8]:
```suggestion
- `deploy.sh`, a shell script that helps deploy the scenario in a Kubernetes cluster.
```

File: tests/blackbox/README.md [hunk 5]
```diff
@@ -2,9 +2,17 @@
...
+- `deployment.yml`, a Kubernetes manifest, that is utilised by the aforementioned `deploy.sh`.
```
Comment (mmitoraj) [inline tests/blackbox/README.md:9]:
```suggestion
- `deployment.yml`, a Kubernetes manifest used by the `deploy.sh` shell script.
```

**Review by mmitoraj — APPROVED** [review]

**Pull request merged**